### ISSUE TITLE
Question: Custom command /checkblocks

### ISSUE THREAD

**Issue by realKSMC** [Valandur/Web-API#15]
Is it possible to create an command called `/checkblocks X1 Y1 Z1 X2 Y2 Z2` which responds each block with its minecraft ID, like `/checkblocks 100 100 100 102 100 100` - result: `100 100 100 minecraft:air, 101 100 100 minecraft:stone:2, 102 100 100 minecraft:air` (which is air, polished granite, air). Its an beginning for an decaying world, thats the idea behind if I can check each block. Vanilla minecraft's `/testforblock` does not return exactly the ID, just **** :)

WebAPI would be fantastic for it ;)

**Comment by Valandur** [comment]
This is already possible using the `/block` endpoint. E.g.: `/block/{world-uuid}/{x}/{y}/{z}` will return a single block. To get multiple blocks use: `/block/{world-uuid}/{minX}/{minY}/{minZ}/{maxX}/{maxY}/{maxZ}`

**Comment by realKSMC** [comment]
I'll give it a try, thx for that moment ;)

**Comment by realKSMC** [comment]
I am not really have notion of JS, may u can help me solving that little problem ;)
I tried:

```
const request = require("request");
const _ = require("lodash");

const BASE_API_URL = "http://localhost:8085/api";
const HEADERS = {
    "x-webapi-key": "SECRETKEYHEHE",
};

const BODY = {
    "name": "ItsMe",
    "waitTime": 0,
};

request.post({ url: BASE_API_URL + "/block/d8f1b815-851c-41ba-bb27-ab3967b65ea9/40}/77/386}/44/80/389}", json: true, headers: HEADERS, body: BODY }, (err, response, body) => {
    console.log(err);
    console.log(response.statusCode);
    console.log(JSON.stringify(body, null, 2));
});
```

Getting `null 400 undefined`

**Comment by Valandur** [comment]
You would have to do this with a GET request, so something like this:

```
const request = require("request");
const _ = require("lodash");

const BASE_API_URL = "http://localhost:8085/api";
const HEADERS = {
    "x-webapi-key": "SECRETKEYHEHE",
};

request.get({ url: BASE_API_URL + "/block/d8f1b815-851c-41ba-bb27-ab3967b65ea9/40/77/386/44/80/389", json: true, headers: HEADERS }, (err, response, body) => {
    console.log(err);
    console.log(response.statusCode);
    console.log(JSON.stringify(body, null, 2));
});
```

**Comment by realKSMC** [comment]
That quick answer, thx alot ;) I'll check it out in a few minutes.

**Comment by realKSMC** [comment]
U rock mate! That's the best plugin ever :3

**Comment by realKSMC** [comment]
Is there also an way to list all generated world blocks using an endpoint? ;)

**Comment by Valandur** [comment]
I believe requesting all blocks in a world would take very long and overload the server. You can technically make the area very big in the `/block` request, but I believe I set a limit around 100k blocks or so.

**Comment by realKSMC** [comment]
An good idea ;) Is it technically possible to list the chunks which are generated including the coords?

**Comment by realKSMC** [comment]
Pls add an option to disable the block limit (for own risk) in config :)

**Comment by Valandur** [comment]
> Is it technically possible to list the chunks which are generated including the coords?

There is not yet an endpoint for chunks, only blocks. But that sounds like a good idea, I'm going to add it to the next release


> Pls add an option to disable the block limit (for own risk) in config :)

There is already a config option in `/config/webapi/config.conf` called `maxBlockGetSize` which you can change (at your own risk)

**Comment by realKSMC** [comment]
Ya but I changed it to `100000000000000000000000000`, there needs to be an limit outside the conf.

**Comment by Valandur** [comment]
I believe you can set it to 0 to make it unlimited

**Comment by realKSMC** [comment]
Doesn't work :/ But nice try ;) Still `404`. 

> May I get the 404 because the blocks are not generated yet?

**Comment by realKSMC** [comment]
**My code, may it helps:**

```
const request = require("request");
const _ = require("lodash");

const BASE_API_URL = "http://localhost:8085/api";
const HEADERS = {
    "x-webapi-key": "ADMINHEHE",
};

request.get({ url: BASE_API_URL + "/block/d8f1b815-851c-41ba-bb27-ab3967b65ea9/-5000/0/-5000/5000/255/5000", json: true, headers: HEADERS }, (err, response, body) => {
    console.log(err);
    console.log(response.statusCode);
    console.log(JSON.stringify(body, null, 2));
});
```

**Comment by Valandur** [comment]
Can you also post the full output that you get in the console?

**Comment by realKSMC** [comment]
```
null
404
undefined
```

**Comment by realKSMC** [comment]
Just btw for the chunk endpoint. Maybe add to generate chunks in an defined area where chunks aren't created yet.

**Comment by realKSMC** [comment]
Maybe WebAPI does not know to difference if an block is out of the world? And as result it comes to `404`.

**Comment by Valandur** [comment]
A 404 usually means that the world was not found. Are you sure the UUID is right?
You can change the line
```console.log(response.statusCode);```
to
```console.log(response.statusCode + " - " + response.statusMessage);```
and it should show you a more detailed error message

**Comment by realKSMC** [comment]
Yes I never changed the UUID, also when the error message say exactly what u said. Is it possible like I said if the blocks are not generated yet it comes to that error?

```
null
404 - Could not get world
undefined
```

**Comment by Valandur** [comment]
I believe if the blocks/chunks aren't generated yet then it just returns "air" blocks.
That error sounds like the world isn't loaded. You could check in the admin panel to see if the world is actually loaded. Are there any error messages in the server console when you try to get the blocks?

**Comment by realKSMC** [comment]
Its definetly no "world" bug or something.

`"/block/d8f1b815-851c-41ba-bb27-ab3967b65ea9/-100/0/-100/100/255/100"` works great, just changed the coords.

**Comment by Valandur** [comment]
Oh then this sounds like your server is running out of memory. How much memory does the java process have? And is there any error in the server console?

**Comment by realKSMC** [comment]
The process do not even starts if I say `"/block/d8f1b815-851c-41ba-bb27-ab3967b65ea9/-5000/0/-5000/5000/255/5000"`. The server got 10G of RAM and just uses around 1G.

**Comment by Valandur** [comment]
I meant an error in your minecraft server console? Like where you run normal minecraft commands and stuff
